src/jit/compiler.cpp

```diff
@@ -2396,16 +2396,115 @@ void Compiler::compSetProcessor()
 #endif // _TARGET_XARCH_
 
 #if defined(_TARGET_ARM64_)
-    // There is no JitFlag for Base instructions handle manually
-    opts.setSupportedISA(InstructionSet_Base);
-    // Dummy ISAs for simplifying the JIT code
-    opts.setSupportedISA(InstructionSet_Vector64);
-    opts.setSupportedISA(InstructionSet_Vector128);
-#define HARDWARE_INTRINSIC_CLASS(flag, jit_config, isa)                                                                \
-    if (jitFlags.IsSet(JitFlags::flag) && JitConfig.jit_config())                                                      \
-        opts.setSupportedISA(InstructionSet_##isa);
-#include "hwintrinsiclistArm64.h"
+    if (JitConfig.EnableHWIntrinsic())
+    {
+        // Dummy ISAs for simplifying the JIT code
+        opts.setSupportedISA(InstructionSet_Base);
+        opts.setSupportedISA(InstructionSet_Base_Arm64);
+        opts.setSupportedISA(InstructionSet_Vector64);
+        opts.setSupportedISA(InstructionSet_Vector128);
+    }
+
+    if(jitFlags.IsSet(JitFlags::JIT_FLAG_HAS_ARM64_ATOMICS) && JitConfig.EnableArm64Atomics())
```

Comment:
All of these were pre-existing. I don't think we even support most of them today.